crates/codegen/parser/runtime/src/query_model.rs

```diff
@@ -0,0 +1,144 @@
+use std::rc::Rc;
+
+use super::kinds::{FieldName, RuleKind, TokenKind};
+
+#[derive(Clone)]
+pub enum Query {
+    Binding(Rc<BindingMatch>),
+    Node(Rc<NodeMatch>),
+    Optional(Rc<OptionalMatch>),
+    Alternatives(Rc<AlternativesMatch>),
+    Sequence(Rc<SequenceMatch>),
+    OneOrMore(Rc<OneOrMoreMatch>),
+    Ellipsis,
```

Comment:
What do you think of using the actual semantics (similar to other variants) rather than the syntax kind? for example:

```suggestion
    ZeroOrMoreNodes,
```

Reply:
Well, Ellipsis is a semantic operator. I think you are meaning the implementation semantics, and in this case I think Ellipsis is better, but it's a minor preference over `Anything`